pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -233,29 +235,40 @@ def __validate_init_configuration(self):
                     ' You can save the last checkpoint with ModelCheckpoint(save_top_k=None, monitor=None)'
                 )
 
-    def __init_ckpt_dir(self, filepath, save_top_k):
-        self._fs = get_filesystem(filepath if filepath is not None else "")
+    def __init_ckpt_dir(self, filepath, dirpath, filename, save_top_k):
+        if filepath:
+            if (dirpath or filename):
+                raise MisconfigurationException('add some message')
```

Comment:
any good message here pls?

Reply:
or should we do it like if both `filepath` and `(dirpath or filename)` is passed then `filepath` will be ignored and `dirpath and filename` will be used with a warning?

Reply:
I feel it might be better to throw an exception as `filepath` will be deprecated going forward.